### DISCUSSION TITLE
Cluster 2.10.x upgrade and data integrity

### DISCUSSION THREAD

**Discussion by zessx** [nats-io/nats-server#4781] [Q&A] [answered]
We're about to upgrade our cluster (3 servers) to 2.10.x, and I was wondering how it'd ensure data integrity during the upgrade process. Whenever I'll upgrade the first server and relaunch it, do we agree there is no risk at all for the data stored in the two others servers?
I'm assuming RAFT algorithm is fully independant from the Jetstream storage format, and in my mind there should not have any issue at all. Yet, a confirmation from people having a better understanding of this would still be greatly appreciated 😅

**Comment by derekcollison** [comment, marked as answer]
Backup your streams (and optionally consumers). 

When you upgrade, wait for the healthz endpoint to return ok (or internal NATS equivalent) before moving to the next.
That should do it.

We are releasing 2.10.5 at some point today, so might want to wait on that one.
